### DISCUSSION TITLE
How do I increase my VRAM?

### DISCUSSION THREAD

**Discussion by ghost** [utmapp/UTM#3046] [UTM Help (macOS)] [answered]
I would like to increase the amount of VRAM in my VM, but no matter what I try or where I look, there is no option to do so. I would appreciate it if someone could tell me how to increase the VRAM in a VM.

If it is required, I am using the VM from https://mac.getutm.app/gallery/windows-xp.

Configuration:
UTM Version: 2.1.2
OS Version: macOS 11.5
Intel or Apple Silicon?: Apple Silicon

**Comment by conath** [comment]
There isn't currently a way to increase the VRAM in UTM. You can only influence the amount by changing the graphics card. E. g. `QXL` has more VRAM by default than `cirrus` or `vga`.

While this could theoretically be added, it is unclear why this would be useful. QXL already supports high resolutions (with guest support/drivers).

What are you trying to achieve with your Windows XP VM?

**Comment by ghost** [comment]
I was trying to install Microsoft Plus! for Windows XP but upon installing, a warning message appeared telling me I didn't have enough VRAM to use the full featureset of the application.

<img width="1680" alt="Screen Shot 2021-08-24 at 10 01 40 AM" src="https://user-images.githubusercontent.com/72825659/130650697-c3ba5538-fb67-4d16-89ea-a68e6f2626ab.png">

**Comment by conath** [comment]
Oh, interesting. Does changing to QXL graphics solve this? 

Also check out [Windows XP UTM Fan Edition](https://chrisp.cafe/UTM/xp-utm-fan-edition.html) for my recommended XP setup. (which includes QXL graphics)

**Comment by ghost** [comment]
I was already using the qxl-vga display card when I tried this. Although, I will give your VM a try just to see if it works.

**Comment by ghost** [comment]
I tried out your VM, and unfortunately, it still won't let me install all of the features. Having Microsoft Plus! installed isn't that terribly important, but it still would be nice to have the option to adjust your VRAM size seeing as most other VM programs have that option. Perhaps the UTM team could implement the option to adjust your VRAM size in the next version of UTM? It would be very useful.
<img width="1136" alt="Screen Shot 2021-08-24 at 11 35 15 AM" src="https://user-images.githubusercontent.com/72825659/130664618-60b64013-bcb4-48bf-b5d4-de7b1fbdef10.png">

**Comment by conath** [comment]
I can reproduce this in Windows XP using `qxl-vga` and the XP QXL driver. The guest does not recognize the amount of VRAM. 
![qxl-dxdiag-winxp](https://user-images.githubusercontent.com/12073163/130799907-454d497e-58cb-46da-b50b-94b9b6eefe4b.jpg)

I believe there isn't a good solution available for Windows XP. The other graphics options have less VRAM (cirrus has 4mb, not changeable). I would recommend instead to choose a more modern guest OS and run your Plus! software in compatibility mode, if possible.

I would suggest to add a "custom" option to the graphics card menu:
this would show a text field where one can enter a custom line like `-device qxl-vga,vgamem_mb=128`. Alternatively, a "none (advanced)" option could be added and then one would add the line to QEMU arguments. @osy thoughts?

**Comment by ghost** [comment, marked as answer]
I was fiddling around with the gpu options and managed to get Microsoft Plus! fully installed using the ati-vga graphics card, which takes care of that problem. I also hope that a custom gpu option gets added, as it would not only help me, but also many others. Have a great day :).

**Reply by conath** [reply]
I've tried replicating this but I get a black screen when using the `ati-vga` on M1 Mac host. #2683 

Your VM didn't have a black screen even though you're on M1? What is your exact macOS version? Thanks

**Reply by ghost** [reply]
Sorry for responding so late, I wasn't checking github. The version I am currently on is macOS 11.5.2. What's odd about the ati-vga graphics card is that it only displays in 640 × 480 with 16 colors (at least without drivers). Yet it's the one I tried that let me install Microsoft Plus!. It may not be too much of a surprise, since it has 16MB of VRAM, which is what Microsoft Plus! requested for, but other graphics cards with 16MB of VRAM didn't work such as qxl-vga and vmware-svga.

<img width="752" alt="Screen Shot 2021-08-27 at 11 14 20 AM" src="https://user-images.githubusercontent.com/72825659/131164700-b0d1c9ea-cc70-4a91-baba-a82e70829e52.png">

I installed the Plus! software on a 64-bit Windows XP VM using the ati-vga graphics card without any problems. But I decided to boot up your 32-bit Windows XP VM with the ati-vga graphics card and I got the same black screen you had mentioned. Perhaps the ati-vga graphics card only works on 64-bit OSs?

**Reply by balopez83** [reply]
This can actually be done with the vmware-svga graphics option on Windows XP 32bit. 
I had the same issues as everyone else here but figured out that you can switch to that graphics and successfully install Windows XP Plus with minor effort. 

Instructions: 

1. Change to vmware-svga and boot Windows triggering a driver installation. Do Not Close this window. 
2. Windows doesn't have this driver so you need to get the VMware Tools Pre Vista ISO here: https://packages.vmware.com/tools/frozen/windows/
3. Then mount the iso in UTM
4. Create folder at C:\Extract or another of your choosing
5. Open Command Prompt
6. Type D:\setup.exe /A      Or if you mounted the iso on another drive letter use that one. 
7. Tell Vmware where to extract all of the files
8. Once extraction is complete go to the driver installation and tell it to search for drivers here: C:\Extract\VMware Tools\VMware\Drivers\video_xpdm
9. Let it install and complete.
10. Install Windows XP Plus. Graphics will get really bad but it is usable to install. 
11. Shut Down computer after install
12. Set graphics card back to correct one to regain full graphics support
13. Boot Windows and enjoy Windows XP Plus :)

**Comment by startergo** [comment]
I was able to increase the vRam with the [VMQemuVGA](https://github.com/ivanagui2/VMQemuVGA) and using the `Spice QXL GPU`
.
![Screenshot 2024-10-28 at 10 03 11](https://github.com/user-attachments/assets/7c7cd8c4-034b-4492-a7f6-e93ee4e8197f)

It gives me outright 512MB vRAM.
You will have to add [0x01001b36](https://github.com/startergo/VMQemuVGA/blob/master/Info-FB.plist#L32C45-L32C55) to the `IOPCIPrimaryMatch` in the `Info.plist` of the `kext`
![Screenshot 2024-10-28 at 11 27 49](https://github.com/user-attachments/assets/652374ac-e469-4a2e-b3d1-b5dda6a4517f)
